tests/goal-batch-barrier.test.sh

```
@@ -360,6 +367,285 @@ EOF
+  [ "$ordered" = "101" ] || { echo "B17.merging-dropped-from-green FAIL got '$ordered' want 101"; exit 1; }
+  echo "B17 PASS"
+) || { FAIL=$((FAIL + 1)); echo "  FAIL  B17 block exited non-zero"; }
+
+# ---------------------------------------------------------------------------
+# B18 — #289.2 SKILL.md Phase-2c structural: the green-PR merge loop must take
+# the LOWEST green PR (head -n 1) and flip it to the MERGING sentinel, NOT loop
+# over all green PRs dispatching /merge in one async pass.
+# ---------------------------------------------------------------------------
+echo "== B18: Phase 2c dispatches lowest green PR + MERGING sentinel (#289.2 structural) =="
+GOAL_SKILL="$REPO_ROOT/plugins/uberdev/skills/goal-pipeline/SKILL.md"
+assert_grep "$GOAL_SKILL" '_uberdev_goal_batch_green_prs_ordered "\$GOAL_ID" \| head -n 1' "B18.lowest-green-head1"
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)